com.unity.ml-agents/Runtime/DecisionRequester.cs

```diff
@@ -29,37 +29,31 @@ internal class DecisionRequester : MonoBehaviour
         [FormerlySerializedAs("RepeatAction")]
         public bool TakeActionsBetweenDecisions = true;
 
-        /// <summary>
-        /// Whether or not the Agent decisions should start at an offset (different for each agent).
-        /// This does not affect <see cref="DecisionPeriod"/>. Turning this on will distribute
-        /// the decision-making computations for all the agents across multiple Academy steps.
-        /// This can be valuable in scenarios where you have many agents in the scene, particularly
-        /// during the inference phase.
-        /// </summary>
-        [Tooltip("Whether or not Agent decisions should start at an offset.")]
-        public bool offsetStep;
-
         Agent m_Agent;
-        int m_Offset;
 
         internal void Awake()
         {
-            m_Offset = offsetStep ? gameObject.GetInstanceID() : 0;
             m_Agent = gameObject.GetComponent<Agent>();
-            Academy.Instance.AgentSetStatus += MakeRequests;
+            Debug.Assert(m_Agent != null, "Agent component was not found on this gameObject and is required.");
```

Comment:
Should was also (or instead) add `[RequireComponent(typeof(Agent))]` to the class?

Reply:
I briefly thought about doing exactly that and decided not to.  I'll add it.